src/mlpack/core/tree/rectangle_tree/r_plus_tree_split.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file r_plus_tree_split.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Defintion of the RPlusTreeSplit class, a class that splits the nodes of an R
+ * tree, starting at a leaf node and moving upwards if necessary.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_R_PLUS_TREE_SPLIT_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_R_PLUS_TREE_SPLIT_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace tree /** Trees and tree-building procedures. */ {
+
+/**
+ * The RPlusTreeSplit class performs the split process of a node on overflow.
+ *
+ * @tparam SplitPolicyType The class that helps to determine the subtree into
+ *    which we should insert a child node.
+ * @tparam SweepType The class that sweeps a node along an axis.
```

Comment:
Can you document more what the SweepType class needs to do?  I understand the classes you have written now that I have read it, but the documentation here doesn't give very much insight to someone who is just skimming the code.